data-conversion/src/main.rs

```diff
@@ -0,0 +1,171 @@
+use std::env;
+use std::fs::read_to_string;
+use std::fs::File;
+use std::io::{self, Write};
+
+fn main() {
+    let types: Vec<&str> = vec!["binary", "float", "hex"];
+    let args: Vec<String> = env::args().collect();
+    let mut valid = true;
+    if !types.contains(&args[3].as_str()) {
+        eprintln!("{} is not a valid type to convert from", &args[3]);
+        valid = false;
+    }
+    if !types.contains(&args[4].as_str()) {
+        eprintln!("{} is not a valid type to convert from", &args[4]);
+        valid = false;
+    }
+    if valid {
+        convert(&args[1], &args[2], &args[3], &args[4]);
+    }
+}
+
+/// Converts [filepath_get] from type [convert_from] to [convert_to] in [filepath_send]
+fn convert(
+    filepath_get: &String,
+    filepath_send: &String,
+    convert_from: &String,
+    convert_to: &String,
+) {
+    // Create a file called converted.txt
+    let mut converted = File::create(filepath_send).expect("creation failed");
+
+    if convert_to == "binary" {
+        //Convert from hex to binary
+        if convert_from == "hex" {
+            for line in read_to_string(filepath_get).unwrap().lines() {
```

Comment:
It might be a good idea to decide on a strategy for error handling next time we meet. In the meantime though, I agree with Griffin that it would be good to at least have a nice error message if the input file doesn't exist.